From 40be7c880f40147a227d5a40b9282ee5087155fd Mon Sep 17 00:00:00 2001
Message-ID: <40be7c880f40147a227d5a40b9282ee5087155fd.1701297273.git.sharlatanus@gmail.com>
In-Reply-To: <123a0b9b76c24c29184c527dabb652384f7be762.1701297273.git.sharlatanus@gmail.com>
References: <123a0b9b76c24c29184c527dabb652384f7be762.1701297273.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sun, 29 Oct 2023 13:38:49 +0000
Subject: [PATCH v3 33/66] gnu: go-github-com-valyala-fasthttp: Move to (gnu
 packages golang-web).

* gnu/packages/golang.scm (go-github-com-valyala-fasthttp): Move from
here...
* gnu/packages/golang-web.scm: ...to here.

Change-Id: I60676c0204e198c6050724efa5410c5b695f88fb
---
 gnu/packages/golang-web.scm | 41 +++++++++++++++++++++++++++++++++++++
 gnu/packages/golang.scm     | 40 ------------------------------------
 2 files changed, 41 insertions(+), 40 deletions(-)

diff --git a/gnu/packages/golang-web.scm b/gnu/packages/golang-web.scm
index 352c831e41..f59415452f 100644
--- a/gnu/packages/golang-web.scm
+++ b/gnu/packages/golang-web.scm
@@ -11,6 +11,7 @@
 ;;; Copyright © 2023 Hilton Chain <hako@ultrarare.space>
 ;;; Copyright © 2023 Katherine Cox-Buday <cox.katherine.e@gmail.com>
 ;;; Copyright © 2023 Nicolas Graves <ngraves@ngraves.fr>
+;;; Copyright © 2022 Sharlatan Hellseher <sharlatanus@gmail.com>
 ;;;
 ;;; This file is part of GNU Guix.
 ;;;
@@ -760,6 +761,46 @@ (define-public go-github-com-tdewolff-parse-v2
      "This package contains several lexers and parsers written in Go.")
     (license license:expat)))
 
+(define-public go-github-com-valyala-fasthttp
+  (package
+    (name "go-github-com-valyala-fasthttp")
+    (version "1.39.0")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/valyala/fasthttp")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "12qwx0yk7wjj25v4fswgmj28r69gk94kqdmzavca8k9f0yznniz1"))))
+    (build-system go-build-system)
+    (arguments
+     (list
+      #:import-path "github.com/valyala/fasthttp"
+      #:phases
+      #~(modify-phases %standard-phases
+          (replace 'check
+            (lambda* (#:key inputs #:allow-other-keys #:rest args)
+              (unless
+                  ;; Tests hang forever with gccgo.
+                  (false-if-exception (search-input-file inputs "/bin/gccgo"))
+                (apply (assoc-ref %standard-phases 'check) args)))))))
+    (propagated-inputs
+     (list go-github-com-andybalholm-brotli
+           go-github-com-klauspost-compress
+           go-github-com-valyala-bytebufferpool
+           go-github-com-valyala-tcplisten
+           go-golang-org-x-crypto
+           go-golang-org-x-net
+           go-golang-org-x-sys))
+    (home-page "https://github.com/valyala/fasthttp")
+    (synopsis "Provides fast HTTP server and client API")
+    (description
+     "This package provides a Go module @code{fasthttp} which may be used as
+replacement for native @code{net/http} module.")
+    (license license:expat)))
+
 (define-public go-gopkg-in-square-go-jose-v2
   (package
     (name "go-gopkg-in-square-go-jose-v2")
diff --git a/gnu/packages/golang.scm b/gnu/packages/golang.scm
index 1f1ca8694c..100e59e154 100644
--- a/gnu/packages/golang.scm
+++ b/gnu/packages/golang.scm
@@ -11168,46 +11168,6 @@ (define-public go-github-com-valyala-bytebufferpool
 anti-fragmentation protection.")
     (license license:expat)))
 
-(define-public go-github-com-valyala-fasthttp
-  (package
-    (name "go-github-com-valyala-fasthttp")
-    (version "1.39.0")
-    (source
-     (origin
-       (method git-fetch)
-       (uri (git-reference
-             (url "https://github.com/valyala/fasthttp")
-             (commit (string-append "v" version))))
-       (file-name (git-file-name name version))
-       (sha256
-        (base32 "12qwx0yk7wjj25v4fswgmj28r69gk94kqdmzavca8k9f0yznniz1"))))
-    (build-system go-build-system)
-    (arguments
-     (list
-       #:import-path "github.com/valyala/fasthttp"
-       #:phases
-       #~(modify-phases %standard-phases
-           (replace 'check
-             (lambda* (#:key inputs #:allow-other-keys #:rest args)
-               (unless
-                 ;; Tests hang forever with gccgo.
-                 (false-if-exception (search-input-file inputs "/bin/gccgo"))
-                 (apply (assoc-ref %standard-phases 'check) args)))))))
-    (propagated-inputs
-     (list go-golang-org-x-sys
-           go-golang-org-x-net
-           go-golang-org-x-crypto
-           go-github-com-valyala-tcplisten
-           go-github-com-valyala-bytebufferpool
-           go-github-com-klauspost-compress
-           go-github-com-andybalholm-brotli))
-    (home-page "https://github.com/valyala/fasthttp")
-    (synopsis "Provides fast HTTP server and client API")
-    (description
-     "This package provides a Go module @code{fasthttp} which may be used as
-replacement for native @code{net/http} module.")
-    (license license:expat)))
-
 (define-public go-github-com-valyala-tcplisten
   (package
     (name "go-github-com-valyala-tcplisten")
-- 
2.41.0

